staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -102,6 +103,9 @@ func (c *dynamicResourceClient) Create(obj *unstructured.Unstructured, opts meta
 			return nil, err
 		}
 		name = accessor.GetName()
+		if len(name) == 0 {
+			return nil, fmt.Errorf("name is required")
```

Comment:
name can be empty, if we set the `generateName`

Reply:
not when dealing with a subresource

Reply:
@yue9944882 Not sure I follow. Something like this?
```go
		name = accessor.GetName()
		if len(name) == 0 {
			name = accessor.GetGenerateName()
			if len(name) == 0 {
				return nil, fmt.Errorf("name is required")
			}
		}
```

Reply:
oh tricked. it's subresource.